torch/distributed/algorithms/ddp_comm_hooks/powerSGD_hook.py

```diff
@@ -264,6 +266,31 @@ def __init__(
         # Turn on if compression / decompression computation is a bottleneck.
         self.batch_tensors_with_same_shape = batch_tensors_with_same_shape
 
+    def __getstate__(self):
+        r"""
+        Returns a ``PowerSGDState`` which will be pickled and saved.
+        ``process_group`` is not serializable and excluded from
+        a returned state.
+        """
+        slots = copy.copy(self.__slots__)
+        slots.remove("process_group")
+        return {
+            slot: getattr(self, slot)
+            for slot in slots if hasattr(self, slot)
+        }
```

Comment:
Could we do:
```suggestion
        return {
            slot: getattr(self, slot)
            for slot in slots if hasattr(self, slot) and slot != "process_group"
        }
```
I think you are creating a new `list` with `slots = copy.copy(self.__slots__)`, but we can avoid this shallow copy and iterate directly over the existing `self.__slots__`.

Reply:
I agree, removed unnecessary copy from the updated version.